osc-server/src/main/java/org/osc/core/broker/service/securityinterface/BaseSecurityGroupInterfaceService.java

```diff
@@ -54,20 +55,16 @@ protected VirtualSystem validateAndLoad(EntityManager em, SecurityGroupInterface
             throw new VmidcBrokerValidationException("Security group interfaces cannot be created or updated for appliance manager that does not support policy mapping.");
         }
 
-        for(PolicyDto policyDto : dto.getPolicies()){
-            Policy policy = PolicyEntityMgr.findById(em, policyDto.getId());
-
-            if (policy == null) {
-                throw new VmidcBrokerValidationException("Policy with Id: " + policyDto.getMgrPolicyId() + "  is not found.");
-            }
+		// Validate policies
+		Set<Policy> policies = PolicyEntityMgr.findPoliciesById(em, dto.getPolicyIds());
 
-            ApplianceManagerConnector mc = vs.getDistributedAppliance().getApplianceManagerConnector();
-
-            if (!mc.getPolicies().contains(policy)) {
-                throw new VmidcBrokerValidationException("Policy with Name: " + policy.getName()
-                + " is not defined in the manager: " + mc.getName());
-            }
-        }
+		for (Policy policy : policies) {
```

Comment:
Since policies is coming from the DB, it already validated that the policies are right. This additional check that the manager contains the policies seems applicable for all the other places as well.

maybe the PolicyEntityMgr.findPoliciesById should always take the mc id and bake it in into the query?